topics/programming-language/index.md

```diff
@@ -0,0 +1,11 @@
+---
+aliases: programming-languages
+display_name: Programming language
+short_description: A formal language that specifies a set of instructions that can be used to produce various kinds of output.
+topic: programming-language
+wikipedia_url: https://en.wikipedia.org/wiki/Programming_language
+---
+
+A programming language is a formal language that specifies a set of instructions that can be used to produce various kinds of output. Programming languages generally consist of instructions for computers. There are many programming languages in existence, and each has at least one implementation, each of which consists of its own set of programming tools. Some programming languages require computation to be specified in an imperative form (i.e., as a sequence of operations to perform) while other languages use the declarative form (i.e. the desired result is specified, not how to achieve it).
+
+Programming languages can be used to create programs that implement specific algorithms. Programming languages are used in computer programming to implement algorithms. Most programming languages consist of instructions for computers. There are many programming languages in existence, and each has at least one implementation, each of which consists of its own set of programming tools. Some programming languages require computation to be specified in an imperative form (i.e., as a sequence of operations to perform) while other languages use the declarative form (i.e. the desired result is specified, not how to achieve it).
```

Comment:
This second paragraph is entirely repetition of the first, so let's just get rid of it: 

```suggestion
A programming language is a formal language that specifies a set of instructions that can be used to produce various kinds of output. Programming languages generally consist of instructions for computers. There are many programming languages in existence, and each has at least one implementation, each of which consists of its own set of programming tools. Some programming languages require computation to be specified in an imperative form (i.e., as a sequence of operations to perform) while other languages use the declarative form (i.e. the desired result is specified, not how to achieve it).
```

By the way, if you're using ChatGPT or the like to generate these, I'm fine with it - we have plenty of Wikipedia excepts in here - but I would recommend giving them an editorial pass as a human. Generative AI can do some really impressive stuff, but it will occasionally glitch and produce nonsense, or (worse) get things outright wrong but present them with authority.

Reply:
Oops. Using davinci for a starting point, but I definitely think I was on autopilot on this one and missed the HITL